<?xml version="1.0"?>

  <!-- This laucher is like the one with same name in orunav_launch package but
       adding envelope manager and remapping set_task service from 
       coordinator_fake_node and orunav_vehicle_execution_node -->




<launch>
  <!-- TF prefix of robot -->
  <arg name="prefix" default="robot1"/>

  <!-- Enable/disable sensor simulation -->
  <arg name="velodyne" default="false"/> <!-- requires velodyne_simulator -->
  <arg name="kinect2"  default="false"/> <!-- requires kinect2_description -->
  <arg name="kinect1"  default="false"/> <!-- deprecated (only used for pallet detection?) -->

  <!-- Enable/disable software components -->
  <arg name="coordination" default="true"/>
  <arg name="human_perception" default="false"/>

  <!-- Delete humans from scene? -->
  <arg name="delete_humans" default="true" unless="$(arg human_perception)"/>
  <arg name="delete_humans" default="false" if="$(arg human_perception)"/>


  <rosparam command="load" file="$(find orunav_params)/config/global.cfg"/>
  <rosparam command="load" file="$(find orunav_params)/config/citi1_simulation.cfg" />

  <!-- LOAD WORLD -->
  <include file="$(find gazebo_worlds_oru)/launch/ncfm_world.launch"/>

 <include file="$(find cititruck_gazebo)/launch/cititruck_ncfm.launch">
     <arg name="tf_prefix" value="$(arg prefix)"/>
     <arg name="velodyne" value="$(arg velodyne)"/>
     <arg name="kinect2" value="$(arg kinect2)"/>
     <arg name="kinect1" value="$(arg kinect1)"/>
  </include>

  <!--include file="$(find gazebo_worlds_oru)/launch/empty_map_server.launch"/-->
  <include file="$(find gazebo_worlds_oru)/launch/ncfm_map_server.launch"/>
  <!--include file="$(find orunav_mpc)/launch/citi1_simulation.launch"/-->

  <node pkg="orunav_vehicle_execution" type="point_n_click_target_client" name="point_n_click_target_client_citi1" output="screen">
    <param name="load_operation" value="false" />
    <rosparam param="robot_ids">[1]</rosparam>
    <param name="multiple_vehicles" value="true" />
  </node>

  
  <node pkg="envelope_manager" type="em_node" name="envelope_manager_node" output="screen">  
    <rosparam param="/use_ct">false</rosparam>
    <remap from="/execute_task_coordinator" to="/$(arg prefix)/execute_task_coordinator"/>
    <remap from="/execute_task" to="/$(arg prefix)/execute_task"/>
  </node>


  <!-- HERE -->
  <node pkg="orunav_coordinator_fake" type="coordinator_fake_node" name="coordinator_fake_node" output="screen">
    <param name="multiple_vehicles" value="true" />
    <param name="use_ct" value="false" />  
    <remap from="/$(arg prefix)/execute_task" to="/$(arg prefix)/execute_task_coordinator"/>
  </node>

  <!-- HUMAN PERCEPTION MASTER -->
  <node name="delete_humans_from_gazebo" pkg="gazebo_models_oru" type="delete_humans_from_gazebo.py" output="screen" if="$(arg delete_humans)"/>
  
  <group if="$(arg human_perception)">
    <include file="$(find iliad_human_perception_launch)/launch/master.launch">
      <!-- Publish groundtruth tracked persons from Gazebo -->
      <arg name="simulation" value="true"/>
    </include>
  </group>
  

  <!-- ALL ROBOT SPECIFIC TASKS -->
  <!--include file="$(find orunav_launch)/launch/motion_planning_and_control_smp.launch"-->
  <include file="$(find orunav_launch)/launch/motion_planning_and_control.launch">
    <arg name="prefix" value="robot1"/>
    
  </include>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find cititruck_description)/launch/iliad_cititruck.rviz"/>

</launch>
